### PR TITLE
Show hints for interactive components on SceneView and GameView (only if Gizmoz enabled)

### PR DESCRIPTION
**Pull request by Kuniwak**
I implemented hints for interactive components on SceneView and GameView. Related #45.

### Limitations

Hints for components on canvas that has screen space overlay as the render mode are not rendered on GameView. There are on only SceneView.

### CODE REVIEW

**Review by nowsprinting — COMMENTED** [review]
I have a few minor comments.

Also, it might be more beneficial to add the following menu item.
```
public static class HintComponentMenu
{
    [MenuItem("GameObject/TestHelper.Monkey/Interactive Component Hint Object")] // on Hierarchy window
    private static void CreateInteractiveComponentHintObjectMenuItem()
    {
        new GameObject("Interactive Component Hint").AddComponent<InteractiveComponentHint>();
    }
}
```

File: Runtime/Hints/InteractiveComponentHint.cs [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+            if (_count == 0)
+            {
+                Refresh();
+                _count = framesPerRefresh;
+            }
+            else
+            {
+                _count--;
+            }
+        }
```
[the remark this answers is not in the document]
Comment (Kuniwak) [reply Runtime/Hints/InteractiveComponentHint.cs]: My implementation was based on obsolete one that implmeneted when I aimed showing hints though not playing. Using `Time.frameCount` is better.

File: Tests/Runtime/TestDoubles/SpyOnPointerDownUpHandler.cs [hunk 2]
```diff
@@ -10,7 +10,6 @@ namespace TestHelper.Monkey.TestDoubles
     /// <summary>
     /// Pointer down/up event handler
     /// </summary>
-    [AddComponentMenu("")] // Hide from "Add Component" picker
```
[the remark this answers is not in the document]
Comment (Kuniwak) [reply Tests/Runtime/TestDoubles/SpyOnPointerDownUpHandler.cs]: I reverted at 820f12a.

File: Runtime/Hints/InteractiveComponentHint.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (Kuniwak) [reply Runtime/Hints/InteractiveComponentHint.cs]: Fixed at [65e2def](https://github.com/nowsprinting/test-helper.monkey/pull/53/commits/65e2defe944bd9b0f9096e2824f0c062efc72081) and 
[52542f8](https://github.com/nowsprinting/test-helper.monkey/pull/53/commits/52542f8daaa581ed9c6f69d2c0e6236ffc6f8c58)

File: Runtime/Hints/InteractiveComponentHint.cs [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+        /// Color for interactive components that users can operate
+        /// </summary>
+        [SerializeField]
+        public Color activePointColor = Color.blue;
+
+        /// <summary>
+        /// Color for interactive components that users cannot operate
+        /// </summary>
+        [SerializeField]
+        public Color inactivePointColor = Color.red;
```
[the remark this answers is not in the document]
Comment (Kuniwak) [reply Runtime/Hints/InteractiveComponentHint.cs]: Fixed at [0cb407e](https://github.com/nowsprinting/test-helper.monkey/pull/53/commits/0cb407e92f9f5eacac3fd2a04af612c199088bdf)

File: Runtime/Hints/InteractiveComponentHint.cs [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+    /// <summary>
+    /// Component to make operation positions visible
+    /// </summary>
+    public class InteractiveComponentHint : MonoBehaviour
+    {
+        /// <summary>
+        /// Color for interactive components that users can operate
+        /// </summary>
+        [SerializeField]
+        public Color activePointColor = Color.blue;
```
[the remark this answers is not in the document]
Comment (Kuniwak) [reply Runtime/Hints/InteractiveComponentHint.cs]: Fixed at [0cb407e](https://github.com/nowsprinting/test-helper.monkey/pull/53/commits/0cb407e92f9f5eacac3fd2a04af612c199088bdf)

File: Editor/UI/Hints/InteractiveComponentHintEditor.cs [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    {
+        private static readonly string s_activePointColor = L10n.Tr("Active Point");
+
+        private static readonly string s_activePointTooltip =
+            L10n.Tr("Color for interactive components that user can operate");
+
+        private SerializedProperty _activePointColorProp;
+        private GUIContent _activePointColorContent;
+
+        private static readonly string s_inactiveColor = L10n.Tr("Inactive Point");
```
[the remark this answers is not in the document]
Comment (Kuniwak) [reply Editor/UI/Hints/InteractiveComponentHintEditor.cs]: Fixed at [0cb407e](https://github.com/nowsprinting/test-helper.monkey/pull/53/commits/0cb407e92f9f5eacac3fd2a04af612c199088bdf)

File: Editor/UI/Hints/InteractiveComponentHintEditor.cs [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+namespace TestHelper.Monkey.Editor.Hints
+{
+    /// <summary>
+    /// Custom editor for <c cref="InteractiveComponentHint" />
+    /// </summary>
+    [CustomEditor(typeof(InteractiveComponentHint))]
+    public class InteractiveComponentHintEditor : UnityEditor.Editor
+    {
+        private static readonly string s_activePointColor = L10n.Tr("Active Point");
```
[the remark this answers is not in the document]
Comment (Kuniwak) [reply Editor/UI/Hints/InteractiveComponentHintEditor.cs]: Fixed at [0cb407e](https://github.com/nowsprinting/test-helper.monkey/pull/53/commits/0cb407e92f9f5eacac3fd2a04af612c199088bdf)

**Comment by Kuniwak** [comment]
I added you suggested to context menu at [2ae2179](https://github.com/nowsprinting/test-helper.monkey/pull/53/commits/2ae217912d26b1d7c1bfa2c8b0f0319e0cb2cc83)

**Comment by Kuniwak** [comment]
I changed the default color for "Corrected point (reachable)" to blue from yellow you suggested. Red and orange and yellow are warning colors. The color for points monkey operators can operate should not be a warning color.

**Comment by nowsprinting** [comment]
> The color for points monkey operators can operate should not be a warning color.

Right. Thank you.

**Review by nowsprinting — APPROVED** [review]
LGTM, Thank You!

**Pull request merged**